src/widgets/MediaPlayer.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import MqttClientSingleton from '../util/MqttClientSingleton'
+import './SwitchOnOff.css';
+import './MediaPlayer.css';
+
+class MediaPlayer extends Component {
+
+  constructor(props) {
+    super(props);
+    this.mqtt = new MqttClientSingleton();
+  }
+
+  handleClick = (event) => {
+    if (this.props.readOnly) {
+      return
+    }
+    const message = {
+      command: 'switchlight',
+      idx: this.props.idx,
+      switchcmd: this.props.value === 0 ? 'On' : 'Off',
+      level: 100
+    };
+    this.mqtt.publish(message);
+  }
+
+  getButtonStyle() {
+    const theme = this.props.theme;
+    if (!theme) {
+      return {};
+    }
+    const style = {
+      color: theme.textOn,
+      background: theme.buttonOn
+    }
+    if (this.props.valueText === "Off" || this.props.value === 0) {
+      style.color = theme.textOff;
+      style.background =  theme.buttonOff;
+    }
+    return style;
+  }
+
+  render() {
+    const valueText = this.props.valueText || (this.props.value === 0 ? 'Off' : 'On');
+    const valuePlaying = ((this.props.playing === "100" && valueText === "On") ? 'Loading...' : ((this.props.playing === "100" && valueText === "Off") ? 'Off' : this.props.playing));
+    return (
+	  <button className="switch" style={this.getButtonStyle()} onClick={this.handleClick} title={valueText}>{this.props.label}<br/><i className="playing">{valuePlaying}</i></button>
```

Comment:
Ah I see, sorry. I meant the br tag... it gets parsed by github's comment system when I type it with the chevrons :)